app/src/main/java/com/abhishekgupta/githubsearch/viewmodel/SearchViewModel.kt

```diff
@@ -27,16 +26,15 @@ class SearchViewModel(
         viewModelScope.launch {
             userLiveData.value = Resource.Loading()
 
-            val userDeferred = async { repo.getUser(userName) }
+            val userDeferred = repo.getUser(userName)
 
-            val followers = async { repo.getUserFollowers(userName) }
+            val followers = repo.getUserFollowers(userName)
 
-            val following = async { repo.getUserFollowing(userName) }
```

Comment:
Thank you for your suggestion.
I thought to make this calls in parallel, hence async used here. removing async will make calls in serial.
what are your thoughts on this ?

Reply:
this call will be asynchronous because it is executed in a viewModelScope.launch block

Reply:
Hey @Georg I was talking about parallel calls.

Using async repo/api calls will be executed in parallel and will take less time to inflate the data on UI.

If we remove async all this calls will me made in serial and hence more time the inflate the data on UI.

Time Taken when async is used

Time taken for request : 1226
Time taken for request : 1124

Time Taken when async is removed

Time taken for request : 2215
Time taken for request : 1866

Reply:
we call a suspend function from viewModel that is executed asynchronously

Reply:
you have a suspended function specified in the repository. it cannot be executed non-asynchronously